### PR TITLE
feat(workspace/app): add resource for publishing application

### PR DESCRIPTION
**Pull request by wuzhuanhong**
**What this PR does / why we need it**:
Add new resource used to manage application publishment.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. add new resource.
2. add corresponding to docunment and acceptance test.
```

## PR Checklist

* [x] Tests added/passed.

```
./scripts/coverage.sh -o workspace -f TestAccAppPublishment_basic
Prepare to calculate the coverage the following command:
TF_ACC=1 go test "./huaweicloud/services/acceptance/workspace" -v -coverprofile="./huaweicloud/services/acceptance/workspace/workspace_coverage.cov" -coverpkg="./huaweicloud/services/workspace" -run TestAccAppPublishment_basic -timeout 360m -parallel 10
=== RUN   TestAccAppPublishment_basic
=== PAUSE TestAccAppPublishment_basic
=== CONT  TestAccAppPublishment_basic
--- PASS: TestAccAppPublishment_basic (51.34s)
PASS
coverage: 12.5% of statements in ./huaweicloud/services/workspace
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/workspace 51.406s coverage: 12.5% of statements in ./huaweicloud/services/workspace
```

* [x] Documentation updated.
* [x] Schema updated.
* [x] CheckDeleted.

  - **a. During query operation (Read Context)**
    aa. Resource not found
    image
  
    ab. Related resources (parent resources) not found
  image

### CODE REVIEW

**Comment by wuzhuanhong** [comment]
make testacc TEST=./huaweicloud/services/acceptance/workspace TESTARGS='-run TestAccAppPublishment_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/workspace -v -run TestAccAppPublishment_basic -timeout 360m -parallel 4


=== RUN   TestAccAppPublishment_basic
=== PAUSE TestAccAppPublishment_basic
=== CONT  TestAccAppPublishment_basic
--- PASS: TestAccAppPublishment_basic (45.94s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/workspace 45.992s

**Comment by wuzhuanhong** [comment]
func isAuthorizations(inputObjects, ResAuthorizedAccounts []interface{}) []string {
	failedObjects := make([]string, 0)
	for _, v := range inputObjects {
		account := utils.PathSearch("account", v, "").(string)
		accountType := utils.PathSearch("type", v, "").(string)
		isExisit := false
		for _, item := range ResAuthorizedAccounts {
			resObject := utils.PathSearch("account", item, "").(string)
			resObjectType := utils.PathSearch("account_type", item, "").(string)
			if account == resObject && accountType == resObjectType {
				isExisit = true
				break
			}
		}

		if !isExisit {
			failedObjects = append(failedObjects, fmt.Sprintf("%s | %s;", account, accountType))
		}
	}

	log.Printf("failedObjects===%#v", failedObjects)
	return failedObjects
}

File: docs/resources/workspace_app_publishment.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+variable "app_group_id" {}
+variable "app_name" {}
+variable "execute_path" {}
+
+resource "huaweicloud_workspace_app_publishment" "test" {
+  app_group_id   = var.app_group_id
+  name           = var.app_name
+  type           = 3
+  execute_path   = var.execute_path
+}
```
Comment (Lance52259) [inline docs/resources/workspace_app_publishment.md]:
```suggestion
resource "huaweicloud_workspace_app_publishment" "test" {
  app_group_id  = var.app_group_id
  name          = var.app_name
  type          = 3
  execute_path  = var.execute_path
}
```
wrong format

Comment (wuzhuanhong) [reply docs/resources/workspace_app_publishment.md]: fixed

**Pull request merged**